paddle/fluid/prim/api/composite_backward/composite_backward_api.h

```diff
@@ -898,5 +898,51 @@ void erf_grad(const Tensor& x, const Tensor& out_grad, Tensor* x_grad) {
   }
 }
 
+template <typename T>
+void maximum_grad(const Tensor& x,
+                  const Tensor& y,
+                  const Tensor& out_grad,
+                  int axis,
+                  Tensor* x_grad,
+                  Tensor* y_grad) {
+  if (x_grad) {
+    auto x_tmp = cast<T>(greater_than<T>(x, y), out_grad.dtype());
+    auto dx_res = out_grad * x_tmp;
```

Comment:
make sure logic the same with kernel when x equal to y

Reply:
test x = y in https://github.com/PaddlePaddle/Paddle/pull/51568